server.ts

```
@@ -348,6 +392,18 @@ async function bundleCssIcons() {
     console.log(error);
   }
 
+  // ensure the download directory is created
+  await fs.mkdir(path.join(__dirname, "/src/Icon"), { recursive: true });
+  await fs.mkdir(path.join(__dirname, "/src/Spot"), { recursive: true });
+
+  try {
+    const components = await fetchFromFigma();
+    writeReadme(components);
+  } catch (error) {
+    console.error(error);
+    return; //can't continue without the icons
```

I think it is important here that the process exit with a code different from 0. For example pipeline tools would just print a console error and give us a false feeling that everything worked correctly. My recommendation would be to remove the try catch statement all together. 

Side note:
We do actually a bunch of error handling in this file which I don't see as strictly necessary since we are only printing in console and swallow the error. If something goes wrong with the script we should probably let the process exit (the error message and stack trace will be printed out anyway) - what do you think?

---

Agreed. This is a bit of legacy left over from some earlier iterations. I've cleaned up the exception handling a bit, along with making the output a bit more readable as well.